### PR TITLE
Tf 9929/integrate felony schemas

### PR DESCRIPTION
**Pull request by e-papanicolas**
### Ticket(s)

- _[9929](https://airtable.com/appfJZShN8K4tcWHU/pagXdnDYi0tVl4u8R?HjEAY=reclXMCQ2gA9fKUcv)_

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description

This PR works in tandem with [cv-sanity/PR #6 ](https://github.com/clearviction-devs/clearviction-sanity/pull/6) and should **NOT BE MERGED** until that PR has been merged and changes deployed to Sanity Studio.

- updates `utils/sanity.queries.ts` to allow queries for the new felony pages
- updates `utils/sanity.client.ts` to create usable functions that employ the above Sanity queries
- updates imports and function names in `calculator/[slug].tsx` to accommodate changes to variable names in above 2 files
- removes unused functions in `api/revalidate.js` (that correspond to the deleted schemas from the cv-sanity PR)

### Next steps

This PR sets the stage for the rest of the felony calculator implementation. When product is ready to move forward, sprints can start with thinking about some of these considerations:

#### Routing of the pages
```javascript
/calculator/[slug].tsx

/calculator/misdemeanor/[slug].tsx
/calculator/felony/[slug].tsx
```
- reuse the current `[slug].tsx` or create a new file for the felony pages
- if the HEAD should remain in the misdemeanor, or if based on routing it should get its own section, this implementation can be extended easily to a new section now
- how to integrate the felony page paths and page info queries in the `getProps` and `getPaths` functions of the `[slug]` file
- potentially update `utils/sanity.client` and `utils/sanity.queries` to adapt to chosen strategies

#### Components
- if reusing existing calculator components, remove `/midemeanor/` from components and have just `/calculator/components`

#### ISR Revalidate
- in `api/revalidate` the functionality will need to be extended to account for the felony schema. The revalidate file has links and helpful comments, and our wiki has a video on ISR testing in dev environments that will be helpful for working on this

#### Content
- add new content to the Sanity Studio (not a dev task, but testing and helping make sure this goes smoothly)

### Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings in the console
- [x] There are no new linting errors/problems in my code
- [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

File: utils/sanity.client.ts [hunk 1]
```diff
@@ -18,18 +18,40 @@ const sanityClient = (token?: string) => {
...
+interface CalcPageBySlugParams {
   slug: string;
   token?: string;
-}) {
-  return sanityClient(token)?.fetch(calculatorPagesBySlugQuery, { slug });
 }
 
-export async function getCalculatorPagePaths(): Promise<string[] | undefined> {
-  return sanityClient()?.fetch(calculatorPagePaths);
+// `calculatorInfoPage` is the name of the Sanity MISDEMEANOR page, where the visual "title" for
```
Comment (sramalho94) [inline utils/sanity.client.ts:26]: were these comments meant to be separated?

**Review by sramalho94 — CHANGES_REQUESTED** [review]
just a couple small suggestions so far for this PR. I'll keep analyzing and let you know if I find anything else, but so far it looks great

File: pages/api/revalidate.js [hunk 2]
```diff
@@ -11,19 +11,13 @@ export const config = {
 };
 
 const revalidateLookup = {
-  // keys are sourced from clearviction-sanity's repo --> schema folder --> file --> name
+  // keys are sourced from cv-sanity repo -> schema folder -> index -> export schema types
```
Comment (sramalho94) [inline pages/api/revalidate.js:14]: might be more clear to replace the `->` syntax with just giving pathing `/clearviction-sanity/schemas/index.ts`

Comment (e-papanicolas) [reply pages/api/revalidate.js:14]: Hahaha yeah that does seem like a bit easier 😂

File: utils/sanity.client.ts (hunk 1 above)
Comment (e-papanicolas) [reply utils/sanity.client.ts:26]: 😒 there is an eslint rule that lines can't be > 100 chars

Comment (e-papanicolas) [reply utils/sanity.client.ts:26]: I updated it to try and make it less ugly.

File: pages/api/revalidate.js
[the remark this answers is not in the document]
Comment (e-papanicolas) [reply pages/api/revalidate.js:1]: I like it. Since this file is rarely touched, that seems like a good balance to make the file look better.

**Review by sramalho94 — APPROVED** [review]
looks good to go!

**Pull request merged**